File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/openapi/BotOpenApiClientProvider.java
package com.mieai.qqbot.runtime.openapi;

import com.mieai.qqbot.client.QqOpenApiClient;
import com.mieai.qqbot.domain.bot.BotId;

/** Bot-scoped access to the official QQ OpenAPI client without exposing AppSecret values. */
public interface BotOpenApiClientProvider extends AutoCloseable {
    QqOpenApiClient clientFor(BotId botId);

    void invalidate(BotId botId);

    void invalidateAll();

    @Override
    void close();
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/openapi/BotOpenApiClientProvider.kt
package com.mieai.qqbot.runtime.openapi

import com.mieai.qqbot.client.QqOpenApiClient
import com.mieai.qqbot.domain.bot.BotId

/** Bot-scoped access to the official QQ OpenAPI client without exposing AppSecret values. */
interface BotOpenApiClientProvider : AutoCloseable {
    fun clientFor(botId: BotId): QqOpenApiClient

    fun invalidate(botId: BotId)

    fun invalidateAll()

    override fun close()
}
